rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,391 @@
+- Ensuring that elements fit within a page is not enough. For a responsive design to be successful, the design must also be usable at all screen resolutions and sizes.
+
+### Pros and Cons of Responsive Design
+
+| Pros                          | Cons                                                                                                                                                                                                                                                                                         |
+| ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Abundance of templates to use | Elements can move around fluidly                                                                                                                                                                                                                                                             |
+| Often easier to implement     | Longer mobile download times.                                                                                                                                                                                                                                                                |
+|                               | Tons of `@media` queries often leads to a maintenance overhead in code                                                                                                                                                                                                                       |
+|                               | Images and other high bandwidth features are simply scaled down rather than resized or eliminated outright(unless added some lines of JS which again degrades the performance and increases the mess in code), elements which may not be vital to user experience can cause poor performance |
+
+## Adaptive Design
+
+- While responsive design relies on changing the design pattern to fit the real estate available to it, **adaptive design** has multiple fixed layout sizes. When the site _detects_ the available space, it _selects_ the layout most appropriate for the screen. So, when you open a browser on the desktop, the site chooses the best layout for that desktop screen, resizing the browser has no impact on the design.
+- Adaptive works to detect the screen size and load the appropriate layout for it – generally you would design an adaptive site for six common screen widths(confirm with Saurav once on exact sizes we are doing and update):
+  - 320
```

Comment:
We might replace it with `350px` instead as <1% of devices today are on the `320px` screen size.

Reply:
Yes I want input on this piece from you. we'll catchup and decide what screen sizes we need to support

Reply:
@kamleshchandnani Here are a few **breakpoints** we should have in our system:
- `320px — 480px`: Mobile devices
- `481px — 768px`: Small tablets
- `769px — 1024px`: Small screens, Large Tablets
- `1025px — 1200px`: Desktops
- `1200px and above` —  Extra large screens, TVs

Reply:
Can we add these details to the RFC, look like arbitrary break points without that

Reply:
@saurav12 while defining do you think we'll have to mention for 1201 and above? If yes we need a deterministic breakpoint. This is what I have for now

```
s:  480px // Mobile devices
m:  768px // Large mobile devices
l: 1024px // Small screens, Tablets
xl: 1200px // Desktops
```